### ISSUE TITLE
High cpu usage on clickhouse pod

### ISSUE THREAD

**Issue by sati-max** [SigNoz/charts#381]
Hi

We are using SigNoz in k8s env and we notice that the clickhouse pod has high CPU utilization that doesn't correspond with increased data being send to SigNoz (the data flow is rather constant during the period of time when clickhouse CPU is very high, going even 100% cpu, k8s worker has 8vcpus and due to clickhouse pod high cpu usage it also uses up to 100% available cpu).

How can I debug what is "eating" that CPU (some query from dashboard?, something is being processed in the background of clickhouse?, some rogue query can't finish and doesn't get properly terminated?) This is the top command from the clickhouse pod:
![image](https://github.com/SigNoz/charts/assets/47517196/2d6e882b-aeef-428f-af19-7e2b28de64aa)

We are using default values from your provided values.yaml file

Or I should direct this issue to Clickhouse developers?

**Comment by srikanthccv** [comment]
How many dashboards and alerts do you have configured?

**Comment by srikanthccv** [comment]
Can you share details about the ingestion volume?

**Comment by sati-max** [comment]
Yes, can you tell me how to check it so that the info is correct and not like my guess on it?

**Comment by srikanthccv** [comment]
Please use this dashboard https://signoz-community.slack.com/archives/C01HWUTP4HH/p1706520502116609?thread_ts=1706126863.291779&cid=C01HWUTP4HH

**Comment by sati-max** [comment]
I'm imported the dashboard. What data I've should send/paste here?

**Comment by srikanthccv** [comment]
Share the accepted spans/metrics/logs rate details and the CPU usage of ClickHouse.

**Comment by sati-max** [comment]
Hi

1st SigNoz setup (no spans)

CPU
![image](https://github.com/SigNoz/charts/assets/47517196/0268a3d1-9cc8-44b3-be58-9917072eeeb8)

Metrics
![image](https://github.com/SigNoz/charts/assets/47517196/e6acc2a6-e038-4b1c-a804-49c691f74634)

Logs
![image](https://github.com/SigNoz/charts/assets/47517196/68a2ea63-edea-47ab-bb46-9de2bb162e91)

Exporter DB write latency
![image](https://github.com/SigNoz/charts/assets/47517196/48c16ac7-30b2-488c-b732-280dbccdf394)
![image](https://github.com/SigNoz/charts/assets/47517196/a4e334fd-8a4a-4a99-b0de-072ff73ad9df)

Exporter DB writes/s
![image](https://github.com/SigNoz/charts/assets/47517196/9cdfafdb-82a5-4b10-808d-03267d2b0eb2)
![image](https://github.com/SigNoz/charts/assets/47517196/4d6c480d-f9ef-4a94-9214-97e3f2fdf4f9)


2nd SigNoz setup (no spans)

CPU
![image](https://github.com/SigNoz/charts/assets/47517196/38f973b2-b894-4265-80b2-1165b1bc778f)

Metrics
![image](https://github.com/SigNoz/charts/assets/47517196/04890b9e-addf-4fb1-a233-2c681ce3cb48)

Logs
![image](https://github.com/SigNoz/charts/assets/47517196/dfb2a520-34c4-4c14-84ce-48722a6c91dc)

Exporter DB write latency
![image](https://github.com/SigNoz/charts/assets/47517196/d6c7ff4e-fb7d-4788-bd3c-5db733d4c39a)
![image](https://github.com/SigNoz/charts/assets/47517196/b738a1d7-5257-47d1-a666-bffebbbd3bc3)

Exporter DB writes/s
![image](https://github.com/SigNoz/charts/assets/47517196/cbba0a71-0771-4f66-b4e4-ad6a8d3de0fb)
![image](https://github.com/SigNoz/charts/assets/47517196/5da694f7-eaab-4fa4-8472-c6e02fa06c0b)


I don't see the option to get clickhouse cpu usage from this dashboard, so I added Exporter DB write latency and Exporter DB writes/s. This is the options in $component:
![image](https://github.com/SigNoz/charts/assets/47517196/dc960501-4aee-4553-aee8-397300b61f92)


Thank you, have a good day

**Comment by sati-max** [comment]
Hi

Any update on this issue?

Additional info:
We also noticed that around 1AM our HV shows drastic drop of CPU usage and then it grows gradually to again drop around 1AM.

![image](https://github.com/SigNoz/charts/assets/47517196/16953c2d-72e6-47c1-8735-17de027f49a9)

Looking inside clickhouse logs there doesn't seem any sort of info that any process (processes) are being restarted, there is only info about dropping empty parts:

One clickhouse pod:
`2024.03.07 23:59:44.294105 [ 621 ] {} <Information> signoz_metrics.samples_v2 (5524edce-ebfa-4ce2-8a5b-2024fe7384d2): Will drop empty part 20240206_16711965_16763541_528`
`2024-03-08T01:00:00.837317172+01:00 2024.03.08 00:00:00.837215 [ 623 ] {} <Information> signoz_metrics.time_series_v4_1day (42aa0288-cee9-4ab8-b539-983eae65956b): Will drop empty part 20240207_1235706_1246713_848`
`2024-03-08T01:00:02.255293820+01:00 2024.03.08 00:00:02.255198 [ 622 ] {} <Information> system.session_log (504a6b73-b33d-43ee-b7bb-6d18a7fb5b1f): Will drop empty part 20240207_59628_59725_18`
`2024-03-08T01:00:02.410201796+01:00 2024.03.08 00:00:02.410120 [ 623 ] {} <Information> system.trace_log (fd303ced-2c16-4a40-a7c3-3bc322f1a7ba): Will drop empty part 20240301_134614_146127_116`
`2024-03-08T01:13:48.462419143+01:00 2024.03.08 00:13:48.459673 [ 621 ] {} <Information> system.zookeeper_log_0 (6fdf0680-73d1-43ba-80e6-c67aa22b3ffe): Will drop empty part 20240207_79658_79729_19`
`2024-03-08T01:14:51.143096264+01:00 2024.03.08 00:14:51.142997 [ 620 ] {} <Information> system.asynchronous_metric_log_2 (e90b1311-448a-4247-880e-e6ebcff51e4e): Will drop empty part 20240207_106338_106418_24`
`2024.03.08 00:17:16.594846 [ 620 ] {} <Information> system.metric_log_3 (54a0f04c-55c9-4888-bb2b-4df991dabcc1): Will drop empty part 20240207_106007_106100_24`
`2024-03-08T01:18:59.543631770+01:00 2024.03.08 00:18:59.543536 [ 605 ] {} <Information> system.query_log_2 (22821b2a-0dd6-4380-be94-ad1141c9fe96): Will drop empty part 20240207_254700_254761_5`
`2024-03-08T01:19:16.422328170+01:00 2024.03.08 00:19:16.422233 [ 624 ] {} <Information> system.part_log_2 (df835ac4-1326-4f2e-94ca-f4a9e9226303): Will drop empty part 20240207_256730_256817_18`


Second clickhouse pod (different signoz)

`2024-03-08T00:59:52.437603466+01:00 2024.03.07 23:59:52.437502 [ 624 ] {} <Information> signoz_metrics.samples_v2 (f23e59e9-9a75-49cb-81e2-4a63b74ccc86): Will drop empty part 20240107_5078376_5192579_1474`
`2024-03-08T00:59:52.722735845+01:00 2024.03.07 23:59:52.722630 [ 624 ] {} <Information> signoz_metrics.samples_v4 (84e67471-6a73-4c25-b1d3-05755a205559): Will drop empty part 20240206_1274890_1385727_1040`
`2024-03-08T01:00:01.742916974+01:00 2024.03.08 00:00:01.742823 [ 623 ] {} <Info
rmation> signoz_metrics.time_series_v4_1day (367054a7-c948-4954-b45c-0fce07e5a765): Will drop empty part 20240207_1385665_1496389_22689`
`2024-03-08T01:00:02.162485336+01:00 2024.03.08 00:00:02.162389 [ 621 ] {} <Information> system.trace_log (05b3f27f-dc0c-4edb-b857-c485efa6e666): Will drop empty part 20240301_408281_419793_63`
`2024-03-08T01:00:02.575463472+01:00 2024.03.08 00:00:02.575374 [ 623 ] {} <Information> system.query_views_log (8c181ce3-a41c-4510-8149-7fef6c5d7c00): Will drop empty part 20240222_315678_327145_649`
`2024-03-08T01:00:02.772456957+01:00 2024.03.08 00:00:02.772357 [ 621 ] {} <Information> system.session_log (2598f85e-8191-4879-91db-b313b386180f): Will drop empty part 20240207_80155_88510_5196`

I don't see anything in crontabs of clickhouse (/etc/crontabs , /etc/periodic). 

The clickhouse pod also isn't being restarted by kubernetes.


Cheers

**Comment by Nageshbansal** [comment]
Hey @sati-max, have you been experiencing similar performance issues lately? Please try the latest version of SigNoz and let us know if the problem persists.

**Comment by sati-max** [comment]
Hey @Nageshbansal - I think the problem doesn't happen anymore in such way that the clickhouse pod just sits at 100%+ of vcpu.

Sadly I lost the dashboards mentioned here https://github.com/SigNoz/charts/issues/381#issuecomment-1928641152 (and going to slack it doesn't show the archived conversation/thread anymore as it was more than 2 years ago) so I can't recheck the dashboards, but our internal monitoring system doesn't alarm us that we have high cpu problems.

If possible could you maybe share those dashboards than I'll check it again.
